SLEE Console should configure RA Entity properties with various type

Same problem as for issue #17 but related to Long properties. More generic fix/enhancement to SMC tool is required in order to support other java Types as well.

Caused by: java.lang.IllegalArgumentException: Value is of type java.lang.String, required type is java.lang.Long
        at javax.slee.resource.ConfigProperties$Property.setValue(ConfigProperties.java:171)
        at javax.slee.resource.ConfigProperties$Property.<init>(ConfigProperties.java:133)
        at org.mobicents.slee.container.management.console.server.PropertiesInfoUtils.toProperties(PropertiesInfoUtils.java:70)
        at

--- a/tools/slee-management-console/server/src/main/java/org/mobicents/slee/container/management/console/server/PropertiesInfoUtils.java
+++ b/tools/slee-management-console/server/src/main/java/org/mobicents/slee/container/management/console/server/PropertiesInfoUtils.java
@@ -22,11 +22,12 @@
 
 package org.mobicents.slee.container.management.console.server;
 
-import javax.slee.resource.ConfigProperties;
-import javax.slee.resource.ConfigProperties.Property;
-
 import org.mobicents.slee.container.management.console.client.ManagementConsoleException;
 import org.mobicents.slee.container.management.console.client.PropertiesInfo;
+
+import javax.slee.resource.ConfigProperties;
+import javax.slee.resource.ConfigProperties.Property;
+import java.lang.reflect.Constructor;
 
 /**
  * 
@@ -58,13 +59,13 @@ public class PropertiesInfoUtils {
       // obtain a value depending on the type of property
       //String value = propertiesInfo.getProperty(key);
       Object value = propertiesInfo.getProperty(key);
-      if (nameAndType[1].equals("java.lang.Integer")) {
-        try {
-          value = Integer.valueOf((String)value);
-        } catch (NumberFormatException e) {
+      try {
+        Class clazz = Class.forName(nameAndType[1]);
+        Constructor con = clazz.getConstructor(String.class);
+        value = con.newInstance((String)value);
+      } catch (Exception e) {
           throw new ManagementConsoleException("Value of " + nameAndType[0]
-                  + " should be java.lang.Integer. " + e.getMessage());
-        }
+                  + " is not supported for type " + nameAndType[1] + ". " + e.getMessage());
       }
 
       Property property = new Property(nameAndType[0], nameAndType[1], value);

